lib/collection/src/shards/local_shard/scroll.rs

```diff
@@ -131,8 +132,19 @@ impl LocalShard {
             })
         };
 
+        let unwrap_vec = |vec: Vec<Result<Vec<ExtendedPointId>, OperationError>>| -> Result<Vec<Vec<ExtendedPointId>>, OperationError> {
+            for ele in vec.iter() {
+                if let Err(err) = ele {
+                    return Err(err.clone());
+                }
+            }
+            Ok(vec.into_iter().map(|x| x.unwrap()).collect())
+        };
+
         let non_appendable = try_join_all(non_appendable.into_iter().map(read_filtered)).await?;
+        let non_appendable = unwrap_vec(non_appendable)?;
         let appendable = try_join_all(appendable.into_iter().map(read_filtered)).await?;
+        let appendable = unwrap_vec(appendable)?;
```

Comment:
We can probably simplify this by collecting like this: `collect::<Result<Vec<Vec<ExtendedPointId>>>()`

It takes out the results short cutting early for us.